dd-sdk-android/src/main/cpp/datadog-native-lib.cpp

```diff
@@ -0,0 +1,95 @@
+//
+// Datadog Native Library.
+//
+#include <jni.h>
+#include <sstream>
+#include <cstring>
+#include <fstream>
+#include "utils/signal-monitor.h"
+#include "utils/backtrace-handler.h"
+#include "datadog-native-lib.h"
+#include "utils/datetime.h"
+#include "utils/fileutils.h"
+
+static const char *STORAGE_DIR;
+static const char *SERVICE_NAME;
+static const char *LOGGER_NAME;
+static const char *GENERIC_MESSAGE;
+static const char *EMERGENCY_STATUS;
+static const char *ENVIRONMENT;
+static const char *ERROR_KIND = "Native";
+
+void onCrashSignalIntercepted(int signal, const char* signalName, const char* errorMessage) {
+    // sync everything
+    static pthread_mutex_t handlerMutex = PTHREAD_MUTEX_INITIALIZER;
+    pthread_mutex_lock(&handlerMutex);
+
+    // create crash reporting directory if it does not exist
+    createDirIfNotExists(STORAGE_DIR);
```

Comment:
Good point...will add make this to return a boolean and in case it fails I will return from this function as nothing will be possible in that moment. Will also send a log inside the Android Logcat in this case to warn the user.